src/swe/physics_swe.c

```diff
@@ -367,16 +490,18 @@ PetscErrorCode RHSFunctionSWE(TS ts, PetscReal t, Vec X, Vec F, void *ctx) {
     }
   }
 
-  // write out debugging info for maximum courant number
+  // find the maximum courant number and write out the debugging info
   if (rdy->config.logging.level >= LOG_DEBUG) {
-    MPI_Allreduce(MPI_IN_PLACE, &courant_num_diags, 1, courant_num_diags_type, courant_num_diags_op, rdy->comm);
+    PetscCall(SWEFindMaxCourantNumber(rdy));
```

Comment:
Should we check to see whether the Courant number is set here, to avoid an unnecessary recalculation?

Reply:
I updated the comment to indicate that we must find the latest Courant number and disregard the previously computed value.